todel/src/models/gateway.rs

```diff
@@ -1,14 +1,127 @@
 use serde::{Deserialize, Serialize};
 
-use super::Message;
+use crate::conf::PandemoniumConf;
 
-/// The Pandemonium Payload Enum
-#[autodoc]
+use super::{InstanceInfo, Message};
+
+/// Pandemonium websocket payloads sent by the server to the client
+#[autodoc(category = "Gateway")]
 #[derive(Debug, Clone, Serialize, Deserialize)]
 #[serde(rename_all = "SCREAMING_SNAKE_CASE")]
 #[serde(tag = "op", content = "d")]
-pub enum Payload {
-    Ping,
+pub enum ServerPayload {
+    /// A [`ClientPayload`] `PING` payload response.
+    ///
+    /// -----
+    ///
+    /// ### Example
+    ///
+    /// ```json
+    /// {
+    ///   "op": "PONG"
+    /// }
+    /// ```
     Pong,
+    /// The event sent when the client gets gateway ratelimited.
+    ///
+    /// The client is supposed to wait `wait` milliseconds before sending any more events,
+    /// otherwise they are disconnected.
+    ///
+    /// -----
+    ///
+    /// ### Example
+    ///
+    /// ```json
+    /// {
+    ///   "op": "RATE_LIMIT",
+    ///   "d": {
+    ///     "wait": 1010 // 1.01 seconds
+    ///   }
+    /// }
+    /// ```
+    RateLimit {
+        /// The amount of milliseconds you have to wait before the rate limit ends
+        wait: u64,
+    },
+    /// The payload sent by the server when you initiate a new gateway connection.
+    ///
+    /// -----
+    ///
+    /// ### Example
+    ///
+    /// ```json
+    /// {
+    ///   "op": "HELLO",
+    ///   "d": {
+    ///     "heartbeat_interval": 45000,
+    ///     "instance_info": {
+    ///       "instance_name": "EmreLand",
+    ///       "description": "More based than Oliver's instance (trust)",
+    ///       "version": "0.3.3",
+    ///       "message_limit": 2048,
+    ///       "oprish_url": "https://example.com",
+    ///       "pandemonium_url": "https://example.com",
+    ///       "effis_url": "https://example.com",
+    ///       "file_size": 20000000,
+    ///       "attachment_file_size": 100000000
+    ///     },
+    ///     "pandemonium_info": {
+    ///       "url": "https://example.com",
+    ///       "rate_limit": {
+    ///         "reset_after": 10,
+    ///         "limit": 5
+    ///       }
+    ///     }
+    ///   }
+    /// }
+    /// ```
+    Hello {
+        /// The amount of milliseconds your ping interval is supposed to be
+        heartbeat_interval: u64,
+        /// The instance's info
+        ///
+        /// This is the same payload you get from the [`get_instance_info`] payload without
+        /// ratelimits
```

Comment:
https://discord.com/channels/980412957060137001/1048626763615449138/1096240851472101508

Reply:
I'm not entirely sure when but I believe we decided to keep it to pandemonium info while working on it in a VC which should provide you with the info you need as to properly interface with the gateway without getting bombarded with god-knows-how-many megabytes of rate-limiting information every time you connect to the gateway